### PR TITLE
Redeploy after variables change

### PR DESCRIPTION
**Pull request by gschier**
This PR adds logic to redeploy the project after setting/deleting variables. It will only redeploy if a deploy exists, and the last deploy wasn't triggered from `railway up`.

NOTE: I originally planned to move the `up` deployment detection to the GQL endpoint but decided against it so the CLI could have more control over the user experience. I think the ideal case down the road is to store the code used to build/deploy and simply re-trigger that with new variables when it changes. That way, we don't have to care about where the previous deploy came from.

## If last deploy was from GH triggers:

Set variables and show loading while deploy is triggering

image

Display success message when complete

image

## If last deploy was from `up`:

image

### CODE REVIEW

**Comment by JakeCooper** [comment]
Deployment URL comes back with the response right? Would be nice to have a click that goes directly to the deployment item (similar to the up response)

**Comment by gschier** [comment]
> Deployment URL comes back with the response right? Would be nice to have a click that goes directly to the deployment item (similar to the up response)

Nah, just a boolean, but we could definitely update it to return that.

**Comment by JakeCooper** [comment]
Should be updated regardless. Will do as part of [RAIL-735](https://linear.app/railway/issue/RAIL-735/return-deployed-url)

File: cmd/variables.go [hunk 1]
```diff
@@ -70,6 +70,42 @@ func (h *Handler) VariablesSet(ctx context.Context, req *entity.CommandRequest)
...
 
+	err = h.maybeRedeployAfterVariablesChange(ctx, environment)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (h *Handler) maybeRedeployAfterVariablesChange(ctx context.Context, environment *entity.Environment) error {
```
Comment (JakeCooper) [inline cmd/variables.go:81]: lol, can we rename this to `validateAndUpdateDeployment` or something like that. `maybe` just doesn't sit right here

**Review by JakeCooper — APPROVED** [review]
Looks good beyond the whole `maybe` thing

File: cmd/variables.go (hunk 1 above)
Comment (gschier) [reply cmd/variables.go:81]: Haha, must be my time with Elm that made me appreciate `maybe`

**Pull request merged**